server/src/guards/role-auth.guard.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { SetMetadata } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import appConfig from 'src/appConfig/configuration';
+
+export enum Role {
+  User,
+  Admin,
+}
+
+// Role decorator
+export const Roles = (roles: Role) => SetMetadata(appConfig('ROLES_KEY'), roles);
+
+@Injectable()
+export class RolesGuard implements CanActivate {
+  constructor(private readonly reflector: Reflector) {}
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    try {
+      const requiredRoles = this.reflector.getAllAndOverride<Role[]>(
+        appConfig('ROLES_KEY'),
```

Comment:
wrong implementation for appConfig file

Reply:
fixed using this pattern 
import { config } from 'dotenv';
config();
export const appConfig = {
  port: process.env.PORT,
  mongodbUrl: process.env.MONGODB_URL,
  jwtSecret: process.env.JWT_SECRET,
  xApiKey: process.env.X_API_KEY,
};
and how it is simply accesible by appConfig.['variable name']